### PR TITLE
Add constant velocity pose predictor

### LINKED ISSUE #149: Write new pose prediction
Because licenses.

### PR DESCRIPTION
**Pull request by mhuzai**
Closes #149.

Draft PR contingent upon response to [this question](https://github.com/rpng/open_vins/issues/247). The response may also motivate a new integrator.

### CODE REVIEW

**Comment by mhuzai** [comment]
My graduation gift to you, @JeffreyZh4ng.

File: pose_prediction/const_vel/plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+     *  \lfloor\mathbf{v}\times\rfloor =
+     *  \begin{bmatrix}
+     *  0 & -v_3 & v_2 \\ v_3 & 0 & -v_1 \\ -v_2 & v_1 & 0
+     *  \end{bmatrix}
+     * @f}
+     *
+     * @param[in] w 3x1 vector to be made a skew-symmetric
+     * @return 3x3 skew-symmetric matrix
+     */
+    static const inline Eigen::Matrix<double, 3, 3> skew_x(const Eigen::Matrix<double, 3, 1> &w) {
```
Comment (jianxiapyh) [inline pose_prediction/const_vel/plugin.cpp:274]: we can move these helper functions into one file. Duplicate across both predictors

**Review by jianxiapyh — COMMENTED** [review]
some minor comment probably can merge without changes

File: configs/monado-mainline.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ plugin_groups:
   - plugin_group:
     - path: offline_imu_cam
     - path: gtsam_integrator
-    - path: pose_prediction
+    - path: pose_prediction/const_vel
```
Comment (Boyuan-Tian) [inline configs/monado-mainline.yaml:5]: Missed a line for const_accel.

File: pose_prediction/const_vel/plugin.cpp (hunk 1 above)
Comment (Boyuan-Tian) [reply pose_prediction/const_vel/plugin.cpp:274]: Not only both predictors but also rk4_integrator and pose_lookup.
Agree to make a unified file for shared util functions.

File: pose_prediction/const_vel/plugin.cpp [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+        const auto curr_R    = quat_2_Rot(Eigen::Matrix<double, 4, 1>{curr_quat.x(), curr_quat.y(), curr_quat.z(), curr_quat.w()});
+        const auto curr_pos  = imu_raw->pos;
+        const auto curr_vel  = imu_raw->vel;
+
+        // Compute new quaternion
+
+        // Option 1: PG's code from issue 92. Same as Propagator.cpp
+        const auto future_q = rot_2_quat(exp_so3(-curr_w * dt) * curr_R);
+
+        // Option 2: PG's formula from issue 92. Same as Trawny eq. 96.
```
Comment (Boyuan-Tian) [inline pose_prediction/const_vel/plugin.cpp:229]: What's the trade-offs between these two options?
Which one should be picked eventually?

File: pose_prediction/const_accel/plugin.cpp [hunk 4]
```diff
@@ -210,10 +205,9 @@ class pose_prediction_impl : public pose_prediction {
     switchboard::reader<switchboard::event_wrapper<time_point>> _m_vsync_estimate;
 	mutable Eigen::Quaternionf offset {Eigen::Quaternionf::Identity()};
 	mutable std::shared_mutex offset_mutex;
-    
 
     // Slightly modified copy of OpenVINS method found in propagator.cpp
-    // Returns a pair of the predictor state_plus and the time associated with the
+    // Returns a pair of the predictor state and the time associated with the
     // most recent imu reading used to perform this prediction.
     std::pair<Eigen::Matrix<double,13,1>,time_point> predict_mean_rk4(double dt) const {
```
Comment (Boyuan-Tian) [inline pose_prediction/const_accel/plugin.cpp:212]: Consider change the function name to `predict_const_accel` to be consistent with `predict_const_vel`.

**Comment by astro-friedel** [comment]
This PR has been inactive for several years.

**Pull request closed without merging**